### DISCUSSION TITLE
Best method for adding support for an app

### DISCUSSION THREAD

**Discussion by packruler** [themepark-dev/theme.park#373] [Q&A] [answered]
I'm interested in adding support for an app or 2 but I'm realizing CSS is not nearly as easy to reverse engineer like code :laughing:. Are there any suggestions, best practices, or standards for this?

**Comment by GilbN** [comment, marked as answer]
Some info here https://github.com/GilbN/theme.park/blob/master/.github/CONTRIBUTING.md

All bases must use the [placeholders css](https://github.com/GilbN/theme.park/blob/master/css/defaults/placeholders.css) and the [transparent css](https://github.com/GilbN/theme.park/blob/master/css/defaults/transparent.css)

Some best practices are (in no particular order) 

 - Try to not use `!important` everywhere unless you actually have to. Like overriding an existing `!important`, or overriding inline styling.
 - The [CSS \[attribute*=value\] Selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors) is golden for wildcard styling where class names are hashed. 
     -  See all the [*arr bases ](https://github.com/GilbN/theme.park/blob/master/css/defaults/servarr-base.css)
     - It can also be used to style ["dark/light theme classes" ](https://github.com/GilbN/theme.park/blob/master/css/base/vuetorrent/vuetorrent-base.css) `[class*="theme--"]` will style both `theme--dark` and `theme--light`

Other than that just look at how the other bases are made.

**Reply by packruler** [reply]
Awesome thank you!

**Reply by GilbN** [reply]
Np. There's also a discord btw https://discord.com/invite/HM5uUKU
